### PR TITLE
First version of spring bean init processor has been added

### PR DESCRIPTION
**Pull request by cosmiclavaflow**

### CODE REVIEW

File: src/main/java/com/academy/ApplicationRunner.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Slf4j
+@Component
+public class ApplicationRunner {
+
+    @Autowired
```
Comment (Taniann) [inline src/main/java/com/academy/ApplicationRunner.java]: for future read about DI with lombok
https://www.baeldung.com/spring-injection-lombok

File: src/main/java/com/academy/config/MainContextConfig.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.academy.ApplicationRunner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan(basePackages = {"com.academy"})
+public class MainContextConfig {
```
Comment (Taniann) [inline src/main/java/com/academy/config/MainContextConfig.java]: it is not config
it is start point with method main
remember about s - single respinsibility for class

File: src/main/java/com/academy/entity/RootFile.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Data
+@Slf4j
+@Component
+@Scope("singleton")
```
Comment (Taniann) [inline src/main/java/com/academy/entity/RootFile.java]: singleton is a default scope for bean
so it is a redundant annotation here

File: src/main/java/com/academy/ApplicationRunner.java (hunk 1 above)
Comment (cosmiclavaflow) [reply src/main/java/com/academy/ApplicationRunner.java]: It's really curious and useful material) I'll try to use it. Thank you

File: src/main/java/com/academy/ApplicationRunner.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    private RootFileTerminal terminal;
+
+    public void start() {
+        System.out.println("List of users: " + userGenerator.getUsers().toString());
+        System.out.println("Do we have access to RootFile? " + terminal.getAccess("root", "1234"));
+    }
+
+    @PostConstruct
+    private void onInit() {
```
Comment (Taniann) [inline src/main/java/com/academy/ApplicationRunner.java]: firstly declare public methods then private 
[Ordering of class contents](https://mate-academy.github.io/style-guides/java/java.html#s3.4.2-ordering-class-contents)

File: src/main/java/com/academy/config/MainContextConfig.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+    @Bean
+    public CustomBeanPostProcessor customBeanPostProcessor() {
```
Comment (Taniann) [inline src/main/java/com/academy/config/MainContextConfig.java]: why did you declare it twice : with annotation Component and Bean
read difference https://stackoverflow.com/questions/51630951/what-is-the-difference-between-bean-annotation-and-component-annotation-at-spr

File: src/main/java/com/academy/service/RootFileTerminal.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Slf4j
+@Service
+@Scope("prototype")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```
Comment (Taniann) [inline src/main/java/com/academy/service/RootFileTerminal.java]: you can write just @RequiredArgsConstructor without pointing onConstructor = @__(@Autowired)

**Pull request merged**